doc/vlan_stacking/vlan_stacking_HLD.md

```diff
@@ -0,0 +1,862 @@
+attr.value.s32 = SAI_VLAN_STACK_MATCH_TYPE_OUTER;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_ORIGINAL_VLAN_ID;
+attr.value.u16 = 20;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_PORT;
+attr.value.oid = 30;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_APPLIED_VLAN_ID;
+attr.value.u16 = 10;
+vlan_stacking_entry_attrs.push(attr);
+
+sai_vlan_api->create_vlan_stack(&vlan_stacking_oid, gSwitchId, (uint32_t)vlan_stacking_entry_attrs.size(), vlan_stacking_entry_attrs.data());
+```
+
+# Configuration and management
+
+## CLI/YANG model Enhancements
+
+### CLI
+
+Add VLAN stacking rule on a physical interface or PortChannel
+
+```
+config vlan-stacking add <interface_name> <s_vlanid> <c_vlanids> <s_vlan_priority>
+  * Append service vlan id with priority for ingress packet when matched customor vlan id.
+    Strip service vlan id for egress packet.
+  * interface_name  : The name of physical interface or PortChannel, e.g. "Ethernet0" or "PortChannel0001".
+  * s_vlanid        : The VLAN ID which will be pushed for ingress direction and popped for egress direction(The range is from 1 to 4094).
+  * c_vlanids       : The VLAN ID list which is used to match on ingress direction(The range is from 1 to 4094).
+  * s_vlan_priority : The priority of Service VLAN.(The range is from 0 to 7).
+```
+
+The format is in the following
+
+```
+admin@sonic:~$ sudo config vlan-stacking add -h
+Usage: config vlan-stacking add [OPTIONS] <interface_name> <s_vlanid> <c_vlanids> <s_vlan_priority>
+
+  Add vlan stacking rule, examples:
+  - config vlan-stacking add Ethernet0 200 100 0
+  - config vlan-stacking add Ethernet0 200 100,110-120 1
+
+Options:
+  -?, -h, --help  Show this message and exit.
+```
+Delete VLAN stacking rule from a physical interface or PortChannel
+
+```
+config vlan-stacking del <interface_name> <s_vlanid> [<c_vlanids>]
+  * interface_name: The name of physical interface or PortChannel, e.g. "Ethernet0" or "PortChannel0001".
+  * s_vlanid  : The VLAN ID which will be pushed for ingress direction and popped for egress direction(The range is from 1 to 4094).
+  * c_vlanids : The VLAN ID list which is used to match on ingress packet(The range is from 1 to 4094).
+```
+
+The format is in the following
+
+```
+admin@sonic:~$ sudo config vlan-stacking del -h
+Usage: config vlan-stacking del [OPTIONS] <interface_name> <s_vlanid> [<c_vlanids>]
+
+  Delete vlan stacking rule, examples:
+  - config vlan-stacking del Ethernet0 200
+  - config vlan-stacking del Ethernet0 200 115-120
+
+Options:
+  -?, -h, --help  Show this message and exit.
+```
+
+Display VLAN stacking rule configuration
+
+```
+show vlan-stacking
+```
+The display format is in the following
+
+```
+admin@root:~# show vlan-stacking
+Interface    s_vlanid   c_vlanids    s_vlan_priority
+-----------  --------  -----------  ----------------
+Ethernet0         200          100                 0
```

Comment:
If we expect to inherit the C-VLAN priority to S-VLAN as well, how do we show it in the output?

Reply:
@venkatmahalingam  The value is must, you can read in cli spec 
Usage: config vlan-stacking add [OPTIONS] <interface_name> <s_vlanid> <c_vlanids> **<s_vlan_priority>**